analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -47,6 +46,313 @@ public void InstanceReference_SetsNotNull_VB()
+
+                private IEnumerable<int> Untracked() => [ 1, 2, 3 ];
+
+                private static void Tag(string name, object arg) { }
+            }
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+        validator.TagValue("Before1").Should().BeNull();
+        validator.TagValue("After1").Should().BeNull();
+        validator.TagValue("Before2").Should().BeNull();
+        validator.TagValue("After2").Should().BeNull();
+    }
+
+    [TestMethod]
+    public void Invocation_DoesntSetNotNull_OnExtensionsNonLinq()
+    {
+        var code = $$"""
+            using System;
+            using System.Linq;
+            using System.Collections.Generic;
+            public class Sample
+            {
+                public void Main(IEnumerable<int> collection)
+                {
+                    Tag("Before1", collection);
+                    _ = collection.NonLinqExtensionNonNullChecking();
+                    Tag("After1", collection);
+
+                    collection = Untracked();
+                    Tag("Before2", collection);
+                    _ = collection.NonLinqExtensionNullChecking();
+                    Tag("After2", collection);
+                }
+
+                private IEnumerable<int> Untracked() => [ 1, 2, 3 ];
+
+                private static void Tag(string name, object arg) { }
+            }
+
+            public static class MyExtensions
+            {
+                public static int NonLinqExtensionNonNullChecking(this IEnumerable<int> source) => 42;
+                public static int NonLinqExtensionNullChecking(this IEnumerable<int> source)
+                {
+                    _ = source ?? throw new ArgumentNullException(nameof(source));
+                    return 42;
+                }
+            }
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+        validator.TagValue("Before1").Should().BeNull();
+        validator.TagValue("After1").Should().BeNull();
+        validator.TagValue("Before2").Should().BeNull();
+        validator.TagValue("After2").Should().BeNull();
+    }
+
+    [DataTestMethod]
+    [DataRow("Count")]
+    [DataRow("Any")]
+    [DataRow("ToArray")]
+    [DataRow("ToList")]
+    public void Invocation_SetsNotNull_OnSourceOfLinqExtension_VB(string method)
+    {
+        var code = $$"""
+            Imports System
+            Imports System.Linq
+            Imports System.Collections.Generic
+
+            Public Class Sample
+                Public Sub Main(collection As IEnumerable(Of Integer))
+                    Tag("Before1", collection)
+                    Dim i1 = collection.{{method}}()                     ' Call as extension
```

Comment:
Please add variant without `()`, it also invokes the method, if no underlying property with the same name exists